workflow/rules/trim_align_dedup.smk

```diff
@@ -405,42 +370,44 @@ rule bam2bw:
        an associated score, RPGC
     """
     input:
-        bam = get_bam_input,
-        ppqt = get_ppqt_input
+        bam                                 = lambda w: get_bam_input(bam_dir, w, paired_end),
+        ppqt                                = lambda w: get_ppqt_input(ppqt_dir, w, paired_end),
     output:
-        outbw=join(workpath,bw_dir,"{name}.{ext}.RPGC.bw"),
+        outbw                               = join(bw_dir, "{name}.{ext}.RPGC.bw"),
     params:
-        rname="bam2bw",
-        effectivegenomesize=config['references'][genome]['EFFECTIVEGENOMESIZE'],
-        paired_end = paired_end,
-        tmpdir=tmpdir,
-        name = "{name}"
-    threads: int(allocated("threads", "bam2bw", cluster)),
-    envmodules: config['tools']['DEEPTOOLSVER'],
-    shell: """
-    if [ ! -d "{params.tmpdir}" ]; then mkdir -p "{params.tmpdir}"; fi
-    tmp=$(mktemp -d -p "{params.tmpdir}")
-    trap 'rm -rf "${{tmp}}"' EXIT
+        rname                               = "bam2bw",
+        name                                = "{name}",
+        effectivegenomesize                 = config['references'][genome]['EFFECTIVEGENOMESIZE'],
+    threads: 
+        int(allocated("threads", "bam2bw", cluster)),
+    envmodules: 
+        config['tools']['DEEPTOOLSVER'],
+    shell: 
+        """
+        if [ ! -d \"""" + tmpdir + """\" ]; then mkdir -p \"""" + tmpdir + """\"; fi
+        tmp=$(mktemp -d -p """ + tmpdir + """)
```

Comment:
We need to add quotes around the path in the `mktemp` command to avoid word splitting:
![image](https://github.com/user-attachments/assets/6db179f5-305e-494f-8196-746b8942fb78)

Tbh, I am more in favor of how we were setting up the tmpdir before. It was cleaner and easier to read. I am not a big fan of mixing python in the shell directive. I am just worried that someone else looking at this in the future may edit this and make a mistake and then something catastrophic could happen when the cleanup occurs.